src/Uno.Toolkit.UI/Controls/ZoomContentControl/ZoomContentControl.cs

```diff
@@ -0,0 +1,467 @@
+	private bool CanScrollRight() => !ContentBoundsVisibility.HasFlag(BoundsVisibilityFlag.Right);
+
+	private async void UpdateHorizontalScrollBarValue()
+	{
+		HorizontalScrollValue = -1 * HorizontalOffset;
+		await RaiseRenderedContentUpdated();
+	}
+
+	private void IsActiveChanged()
+	{
+		if (ResetZoomAndOffsetWhenInactive && !IsActive)
+		{
+			RemoveOffset();
+			ResetZoom();
+		}
+		if (_scrollH is not null)
+		{
+			_scrollH.Visibility = IsActive ? Visibility.Visible : Visibility.Collapsed;
+		}
+		if (_scrollV is not null)
+		{
+			_scrollV.Visibility = IsActive ? Visibility.Visible : Visibility.Collapsed;
+		}
+	}
+
+	private void UpdateScrollLimits()
+	{
+		if (_presenter?.Content is FrameworkElement fe)
+		{
+			var verticalScroll = Math.Max(0, (fe.ActualHeight * ZoomLevel) - ViewportHeight);
+			var horizontalScroll = Math.Max(0, (fe.ActualWidth * ZoomLevel) - ViewportWidth);
+
+			HorizontalMaxScroll = horizontalScroll / 2;
+			VerticalMaxScroll = verticalScroll / 2;
+
+			HorizontalMinScroll = -1 * HorizontalMaxScroll;
+			VerticalMinScroll = -1 * VerticalMaxScroll;
+		}
+	}
+
+	private void CoerceZoomLevel()
+	{
+		ZoomLevel = Math.Clamp(ZoomLevel, MinZoomLevel, MaxZoomLevel);
+	}
+
+	// Template handling
+	protected override void OnApplyTemplate()
+	{
+		T FindTemplatePart<T>(string name) where T : class =>
+			(GetTemplateChild(name) ?? throw new Exception($"Expected template part not found: {name}"))
+			as T ?? throw new Exception($"Expected template part '{name}' to be of type: {typeof(T)}");
+
+		_presenter = FindTemplatePart<ContentPresenter>(TemplateParts.Presenter);
+		_scrollV = FindTemplatePart<ScrollBar>(TemplateParts.VerticalScrollBar);
+		_scrollH = FindTemplatePart<ScrollBar>(TemplateParts.HorizontalScrollBar);
+
+		ResetOffset();
+		ResetZoom();
+
+		RegisterToControlEvents();
+		RegisterPointerHandlers();
+	}
+
+	// Event handlers
+	private void OnLoaded(object sender, RoutedEventArgs e)
+	{
+		CenterContent();
+	}
+
+	private void OnSizeChanged(object sender, SizeChangedEventArgs args)
+	{
+		UpdateContentBoundsVisibility();
+		if (IsLoaded && AutoZoomToCanvasOnSizeChanged)
+		{
+			FitToCanvas();
+		}
+	}
+
+	private void RegisterToControlEvents()
+	{
+		if (_presenter?.Content is FrameworkElement { } fe)
+		{
+			fe.LayoutUpdated += (s, e) =>
+			{
+				ViewportWidth = fe.ActualWidth;
+				ViewportHeight = fe.ActualHeight;
+
+				UpdateScrollLimits();
+			};
+		}
+
+		if (_scrollV is not null)
+		{
+			_scrollV.Scroll += ScrollV_Scroll;
+		}
+
+		if (_scrollH is not null)
+		{
+			_scrollH.Scroll += ScrollH_Scroll;
+		}
+	}
+
+	private void RegisterPointerHandlers()
+	{
+		PointerPressed -= OnPointerPressed;
+		PointerReleased -= OnPointerReleased;
+		PointerMoved -= OnPointerMoved;
+		PointerWheelChanged -= OnPointerWheelChanged;
+
+		PointerPressed += OnPointerPressed;
+		PointerReleased += OnPointerReleased;
+		PointerMoved += OnPointerMoved;
+		PointerWheelChanged += OnPointerWheelChanged;
+	}
```

Comment:
I feel like those subscriptions can happen once in the constructor, and that's it?